### PR TITLE
ING-591: Update bucket management replica indexes handling

### PR DESCRIPTION
**Pull request by chvck**
Motivation
----------
Creating ephemeral buckets currently requires users to explicitly set replica indexes to false. We should update our handling so that replica indexes defaults to false for ephemeral buckets and true for couchbase buckets.

### CODE REVIEW

File: gateway/dataimpl/server_v1/errorhandler.go [hunk 1]
```diff
@@ -243,6 +243,19 @@ func (e ErrorHandler) NewSearchIndexExistsStatus(baseErr error, indexName string
 	return st
 }
 
+func (e ErrorHandler) NewSearchUUIDMismatchStatus(baseErr error, indexName string) *status.Status {
```
Comment (Westwooo) [inline gateway/dataimpl/server_v1/errorhandler.go:246]: This looks like an unrelated change, does the branch need rebasing?

File: gateway/dataimpl/server_v1/bucketadminserver.go [hunk 2]
```diff
@@ -202,7 +206,7 @@ func (s *BucketAdminServer) CreateBucket(
 				DurabilityMinLevel: minimumDurabilityLevel,
 			},
 			ConflictResolutionType: conflictResolutionType,
-			ReplicaIndexDisabled:   !replicaIndexes,
+			ReplicaIndex:           replicaIndexes,
```
Comment (Westwooo) [inline gateway/dataimpl/server_v1/bucketadminserver.go:209]: nit: the `ReplicaIndexes` and `ReplicaIndex` variable name interchange is a bit confusing

**Review by Westwooo — APPROVED** [review]
lgtm, just looks like some changes for another ticket have been pulled in

File: gateway/dataimpl/server_v1/bucketadminserver.go (hunk 2 above)
Comment (chvck) [reply gateway/dataimpl/server_v1/bucketadminserver.go:209]: I think it's actually named wrong in gocbcorex.
